crates/proof-of-sql/src/base/database/owned_column.rs

```diff
@@ -56,6 +56,23 @@ impl<S: Scalar> OwnedColumn<S> {
         }
     }
 
+    /// Returns element at index as scalar
+    ///
+    /// Note that if index is out of bounds, this function will return None
+    pub(crate) fn scalar_at(&self, index: usize) -> Option<S> {
+        (index < self.len()).then_some(match self {
+            Self::Boolean(col) => S::from(col[index]),
+            Self::SmallInt(col) => S::from(col[index]),
+            Self::Int(col) => S::from(col[index]),
+            Self::BigInt(col) => S::from(col[index]),
+            Self::Int128(col) => S::from(col[index]),
+            Self::Scalar(col) => col[index],
+            Self::Decimal75(_, _, col) => col[index],
+            Self::VarChar(col) => S::from(col[index].clone()),
+            Self::TimestampTZ(_, _, col) => S::from(col[index]),
+        })
+    }
```

Comment:
Would it be better to error here instead of returning ```Option```? Index OOB seems like it could be a critical failure path